ambari-server/src/main/resources/stacks/HDP/2.6/upgrades/config-upgrade.xml

```diff
@@ -53,18 +53,14 @@
       </component>
       <component name="DATANODE">
         <changes>
-          <definition xsi:type="configure" id="hdfs_set_data_transfer_protection"
-                      summary="Enables SASL for authentication of data transfer protocol">
+          <definition xsi:type="configure" id="hdfs_set_data_transfer_protection" summary="Enables SASL for authentication of data transfer protocol">
             <type>hdfs-site</type>
-            <set key="dfs.data.transfer.protection" value="authentication,privacy" if-type="hdfs-site"
-                 if-key="dfs.http.policy" if-value="HTTPS_ONLY"/>
+            <set key="dfs.data.transfer.protection" value="authentication,privacy" if-type="cluster-env" if-key="security_enabled" if-value="true" />
           </definition>
 
-          <definition xsi:type="configure" id="hdfs_set_hadoop_rpc_protection_on_kerberized_cluster"
-                      summary="Encrypting the data transfered between hadoop services and clients">
+          <definition xsi:type="configure" id="hdfs_set_hadoop_rpc_protection_on_kerberized_cluster" summary="Encrypting the data transfered between hadoop services and clients">
             <type>core-site</type>
-            <set key="hadoop.rpc.protection" value="authentication,privacy" if-type="cluster-env"
-                 if-key="security_enabled" if-value="true"/>
+            <set key="hadoop.rpc.protection" value="authentication,privacy" if-type="cluster-env" if-key="security_enabled" if-value="true" />
```

Comment:
These are already protected via a kerberos xsi:security condition, do we need to also have a security_enabled=true check?